service/src/main/java/uk/gov/hmcts/reform/fpl/service/UploadDocumentsService.java

```diff
@@ -0,0 +1,116 @@
+package uk.gov.hmcts.reform.fpl.service;
+
+import lombok.RequiredArgsConstructor;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Service;
+import uk.gov.hmcts.reform.fpl.model.CaseData;
+import uk.gov.hmcts.reform.fpl.model.common.Document;
+import uk.gov.hmcts.reform.fpl.model.common.DocumentMetaData;
+import uk.gov.hmcts.reform.fpl.model.common.DocumentSocialWorkOther;
+import uk.gov.hmcts.reform.fpl.model.common.Element;
+import uk.gov.hmcts.reform.fpl.service.time.Time;
+
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.function.Predicate;
+
+import static uk.gov.hmcts.reform.fpl.utils.ElementUtils.findElement;
+
+@Service
+@RequiredArgsConstructor(onConstructor_ = {@Autowired})
+public class UploadDocumentsService {
+
+    private final Time time;
+    private final DocumentUploadHelper documentUploadHelper;
+
+    public Map<String, Object> updateCaseDetailsWithDocuments(CaseData caseDataBefore,
```

Comment:
Method should just be `updateCaseDocuments` or `updateDocuments` I think. And param's should be reversed to read (caseData, caseDataBefore)

Reply:
updated to `updateCaseDocuments`. Don't mind param's to be reversed but any reason you prefer that?

Reply:
Consistency with other methods that do similar thing. I'd expect caseData current to be first followed by caseDataBefore.